spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java

```diff
@@ -818,19 +817,30 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 		DatastorePersistentEntity<?> persistentEntity =
 				this.datastoreMappingContext.getPersistentEntity(example.getProbeType());
 
-		StructuredQuery.Builder builder = keyQuery ? Query.newKeyQueryBuilder() : Query.newEntityQueryBuilder();
-		builder.setKind(persistentEntity.kindName());
-
-		ExampleMatcherAccessor matcherAccessor = new ExampleMatcherAccessor(example.getMatcher());
-		matcherAccessor.getPropertySpecifiers();
 		LinkedList<StructuredQuery.Filter> filters = new LinkedList<>();
+		NullHandler nullHandler = example.getMatcher().getNullHandler();
 		persistentEntity.doWithColumnBackedProperties((persistentProperty) -> {
 			if (!ignoredProperty(example, persistentProperty)) {
 				Value<?> value = getValue(example, probeEntity, persistentEntity, persistentProperty);
-				NullHandler nullHandler = example.getMatcher().getNullHandler();
 				addFilter(nullHandler, filters, persistentProperty.getFieldName(), value);
 			}
 		});
+		persistentEntity.doWithAssociations((AssociationHandler<DatastorePersistentProperty>) association -> {
+			PersistentPropertyAccessor<?> accessor = persistentEntity.getPropertyAccessor(example.getProbe());
+			DatastorePersistentProperty property = association.getInverse();
+			Object value = accessor.getProperty(property);
+			if (value != null) {
+				Key key = objectToKeyFactory.getKeyFromObject(value,
+						this.datastoreMappingContext.getPersistentEntity(value.getClass()));
+				addFilter(nullHandler, filters, property.getFieldName(), KeyValue.of(key));
+			}
+		});
+
+		StructuredQuery.Builder<?> builder =
+				keyQuery
+						? Query.newKeyQueryBuilder()
+						: Query.newEntityQueryBuilder();
+		builder.setKind(persistentEntity.kindName());
```

Comment:
This block used to be done outside of the callback. Did you move it into the callback for readability? Just curious.

Reply:
yes, just to improve readability